sdc/datatypes/hpat_pandas_dataframe_functions.py

```diff
@@ -935,6 +935,105 @@ def sdc_pandas_dataframe_drop_impl(df, labels=None, axis=0, index=None, columns=
     return func_def, global_vars
 
 
+def _dataframe_reduce_columns_codegen_isna(func_name, func_params, columns, df):
+    results = []
+    joined = ', '.join(func_params)
+    func_lines = [f'def _df_{func_name}_impl({joined}):']
+    ind = df_index_codegen_all(df)
+    for i, c in enumerate(columns):
+        result_c = f'result_{c}'
+        func_lines += [f'  series_{c} = pandas.Series(get_dataframe_data({func_params[0]}, {i}))',
+                       f'  {result_c} = series_{c}.{func_name}()']
+        results.append((columns[i], result_c))
+
+    data = ', '.join(f'"{col}": {data}' for col, data in results)
+    func_lines += [f'  return pandas.DataFrame({{{data}}}{ind})']
+    func_text = '\n'.join(func_lines)
+    global_vars = {'pandas': pandas,
+                   'get_dataframe_data': get_dataframe_data}
+
+    return func_text, global_vars
+
+
+"""
+Example func_text for func_name='isna' columns=('FLOAT', 'INT', 'STRING'):
+
+    def _df_isna_impl(df):
+        series_FLOAT = pandas.Series(get_dataframe_data(df, 0))
+        result_FLOAT = series_FLOAT.isna()
+        series_INT = pandas.Series(get_dataframe_data(df, 1))
+        result_INT = series_INT.isna()
+        series_STRING = pandas.Series(get_dataframe_data(df, 2))
+        result_STRING = series_STRING.isna()
+        return pandas.DataFrame({"FLOAT": result_FLOAT, "INT": result_INT, "STRING": result_STRING},
+                                index = df._index)
+"""
+
+
+def sdc_pandas_dataframe_isna_codegen(df, func_name, params, ser_params):
+    all_params = ['df']
+    df_func_name = f'_df_{func_name}_impl'
+    func_text, global_vars = _dataframe_reduce_columns_codegen_isna(func_name, all_params, df.columns, df)
+    loc_vars = {}
+    exec(func_text, global_vars, loc_vars)
+    _reduce_impl = loc_vars[df_func_name]
+
+    return _reduce_impl
+
+
+def df_index_codegen_all(self):
```

Comment:
Why do we need this function at all? Why can't we always use "index=df._index", i.e. even if it's None it should still be handled by dataframe ctor?